src/Riok.Mapperly/Descriptors/Mappings/UserMappings/UserDefinedNewInstanceGenericTypeMapping.cs

```diff
@@ -32,6 +32,10 @@ ITypeSymbol objectType
 {
     public GenericMappingTypeParameters TypeParameters { get; } = typeParameters;
 
+    public bool DoesTypesSatisfySubstitutionPrinciples(SymbolAccessor symbolAccessor, ITypeSymbol sourceType, ITypeSymbol targetType) =>
```

Comment:
IMO this method should not belong to the mapping itself but more to the mapping builder.